assets/js/src/core/modules/app/error-handler/error-handler.ts

```diff
@@ -0,0 +1,28 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import { ErrorModalService } from '@Pimcore/modules/app/error-handler/services/error-modal-service'
+
+interface IErrorContentProvider {
+  getContent: () => string
+}
+
+const trackError = (data: IErrorContentProvider): never => {
+  const errorContent = data.getContent()
+
+  ErrorModalService.showError(errorContent)
```

Comment:
Really like the overall approach now. :)

One last thing, when we can easily achieve it. 

Maybe we can inverse this dependency here?
So that we can register different services beforehand for the different error types?

That would make it super flexible and would allow us to define the handling for each error type in more detail.